core/src/main/java/nl/nn/adapterframework/doc/ModelBuilder.java

```diff
@@ -0,0 +1,174 @@
+				// Jacoco is a tool for code coverage. To have predictible results,
+				// we omit methods introduced by Jacoco.
+				if(reflect.getName().contains("jacoco")) {
+					continue;
+				}
+				methods.put(reflect.getName(), new AttributeSeed(reflect));
+			}
+			methodsWithInherited = new HashMap<>();
+			methodsWithInherited.putAll(methods);
+			for(Method reflect: clazz.getMethods()) {
+				methodsWithInherited.putIfAbsent(reflect.getName(), new AttributeSeed(reflect));
+			}
+			fullName = clazz.getName();
+			simpleName = clazz.getSimpleName();
+		}
+
+		ElementSeed(final String fullName) {
+			this.fullName = fullName;
+		}
+	}
+
+	public ModelBuilder() {
+		model = new FrankDocModel();
+		model.setGroups(new ArrayList<>());
+		model.setAllElements(new HashMap<>());
+	}
+
+	/**
+	 * @param interfaceName The interface for which we want SpringBean objects.
+	 * @return All classes implementing interfaceName, ordered by their full class name.
+	 */
+	static List<SpringBean> getSpringBeans(final String interfaceName) {
+		Class<?> interfaze = InfoBuilderSource.getClass(interfaceName);
+		if(interfaze == null) {
+			throw new NullPointerException();
+		}
+		if(!interfaze.isInterface()) {
+			throw new IllegalArgumentException("Only retrieve Spring beans from an interface");
+		}
+		Set<SpringBean> unfiltered = InfoBuilderSource.getSpringBeans(interfaze);
+		List<SpringBean> result = new ArrayList<SpringBean>();
+		for(SpringBean b: unfiltered) {
+			if(interfaze.isAssignableFrom(b.getClazz())) {
+				result.add(b);
+			}
+		}
+		return result;
+	}
+
+	static List<ElementSeed> getSelfAndAncestorSeeds(Class<?> clazz) {
+		List<ElementSeed> result = new ArrayList<>();
+		result.add(new ElementSeed(clazz));
+		Class<?> superClass = clazz.getSuperclass();
+		while(superClass != null) {
+			result.add(new ElementSeed(superClass));
+			superClass = superClass.getSuperclass();
+		}
+		return result;
+	}
+
+	FrankDocGroup addGroup(String name) {
+		FrankDocGroup group = new FrankDocGroup(name);
+		group.setElements(new HashMap<>());
+		model.getGroups().add(group);
+		return group;
+	}
+
+	void addElementsToGroup(List<ElementSeed> elementHierarchy, FrankDocGroup group) {
+		List<ElementSeed> reversedSeeds = new ArrayList<>(elementHierarchy);
+		Collections.reverse(reversedSeeds);
+		FrankElement parent = null;
+		for(ElementSeed seed: reversedSeeds) {
+			if(model.getAllElements().containsKey(seed.getFullName())) {
+				parent = model.getAllElements().get(seed.getFullName());
+			}
+			else {
+				parent = createFrankElement(seed, parent);
+				model.getAllElements().put(parent.getFullName(), parent);
+			}
+		}
+		group.getElements().putIfAbsent(parent.getFullName(), parent);
+	}
+
+	FrankElement createFrankElement(ElementSeed seed, FrankElement parent) {
+		FrankElement result = new FrankElement(seed.getFullName(), seed.getSimpleName());
+		result.setParent(parent);
+		result.setAttributes(createAttributes(seed));
+		return result;
+	}
```

Comment:
Ik denk dat het duidelijker / inzichtelijker is als deze code een method asFrankElement() is van ElementSeed,

Reply:
De seeds zijn weg zoals je hebt gevraagd.